language_formatters_pre_commit_hooks/pretty_format_kotlin.py

```diff
@@ -65,13 +67,28 @@ def pretty_format_kotlin(argv: typing.Optional[typing.List[str]] = None) -> int:
     # To workaround this limitation we do run ktlint in check mode only,
     # which provides the expected exit status and we run it again in format
     # mode if autofix flag is enabled
-    check_status, check_output = run_command(
-        "java", *jvm_args, "-jar", ktlint_jar, "--log-level", "debug", "--relative", "--", *_fix_paths(args.filenames)
+    #
+    # Logging must be suppressed here as it goes to stdout. This would
+    # interfere with parsing the output JSON.
+    check_result = subprocess.run(  # nosec: B603 B607
+        [
+            "java",
+            *jvm_args,
+            "-jar",
+            ktlint_jar,
+            "--log-level",
+            "none",
+            "--reporter=json",
+            "--relative",
+            "--",
+            *_fix_paths(args.filenames),
+        ],
+        capture_output=True,
     )
 
     not_pretty_formatted_files: typing.Set[str] = set()
-    if check_status != 0:
-        not_pretty_formatted_files.update(line.split(":", 1)[0] for line in check_output.splitlines())
+    if check_result.returncode != 0:
+        not_pretty_formatted_files.update(item["file"] for item in json.loads(check_result.stdout))
```

Comment:
Please preserve the usage of `run_command` functionality instead of directly calling `subprocess.run`.

```suggestion
    check_status, check_output = run_command(
        "java",
        *jvm_args,
        "-jar",
        ktlint_jar,
        "--log-level",
        "none",
        "--reporter=json",
        "--relative",
        "--",
        *_fix_paths(args.filenames),

    not_pretty_formatted_files: typing.Set[str] = set()
    if check_status != 0:
        not_pretty_formatted_files.update(item["file"] for item in json.loads(check_output))
```

Reply:
That's not possible in the current state because run_command captures both stdout and stderr. However, java and ktlint might output messages there which would interfere with json parsing. run_command would need to be changed to not capture stderr for this case.